### ISSUE TITLE
[DF] Support `SHOW COLUMNS`

### ISSUE THREAD

**Issue by andygrove** [dask-contrib/dask-sql#704]
**Is your feature request related to a problem? Please describe.**

This currently fails with:

```
SHOW COLUMNS is not supported unless information_schema is enabled
```

**Describe the solution you'd like**
Do we want to try and use DataFusion's implementation of `information_schema`? 

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by andygrove** [comment]
@jdye64 I think that the easiest solution here is for Dask SQL to override parsing of the `SHOW COLUMNS` command and then implement its own code to produce the expected results.

If we wanted to use DataFusion's `information_schema` support then we would have to add the DataFusion execution engine as a dependency and we would need to register the user's tables with a `SessionContext` and this would likely cause file access to happen (e.g. reading parquet footers) which seems undesirable.

**Comment by jdye64** [comment]
I agree. I will get working on that parsing logic here in the next few days.

**Comment by ayushdg** [comment]
fixed via #721
